dotnet/Gherkin/Parser.cs

```diff
@@ -75,8 +75,20 @@ public enum RuleType
     [System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
     public partial class Parser<T>
     {
+        public Parser()
+            : this(new AstBuilder<T>())
+        {
+        }
+
+        public Parser(AstBuilder<T> astBuilder)
+        {
+            this.AstBuilder = astBuilder;
+        }
+
         public bool StopAtFirstError { get; set;}
 
+        public AstBuilder<T> AstBuilder { get; }

```

Comment:
```
public string Property { get; }
```

is (as far as I know) equivalent to

```
private readonly string myField;
public string Property { get { return myField; } }
```

But will probably change my version to the version with a `readonly` field because I just realized that I have now idea how closely mono follows the development of C#.